paddle/fluid/pybind/tensor_py.h

```diff
@@ -64,6 +65,31 @@ namespace pybind {
 
 namespace details {
 
+template <typename T>
+class PYBIND11_HIDDEN NumpyAllocation : public memory::Allocation {
+ public:
+  explicit NumpyAllocation(const py::array &arr)
+      : Allocation(const_cast<void *>(arr.data()), sizeof(T) * (arr.size()),
+                   paddle::platform::CPUPlace()),
+        arr_(arr.ptr()) {
+    PADDLE_ENFORCE_NOT_NULL(arr_, platform::errors::InvalidArgument(
+                                      "The underlying PyObject pointer of "
+                                      "numpy array cannot be nullptr"));
+    PADDLE_ENFORCE_NE(
+        arr_, Py_None,
+        platform::errors::PreconditionNotMet(
+            "The underlying PyObject pointer of numpy array cannot be None"));
+    Py_INCREF(arr_);
+  }
+  ~NumpyAllocation() override {
+    py::gil_scoped_acquire gil;
+    Py_XDECREF(arr_);
```

Comment:
Not quite important: since `arr_` is not `nullptr`, `Py_DECREF` is enough. 

`Py_XDECREF(arr_)` is equal to `if (arr_ != nullptr) Py_DECREF(arr_);`

Reply:
done